mat/vector_test.go

```diff
@@ -225,17 +225,17 @@ func TestVecDenseScale(t *testing.T) {
 			want:  NewVecDense(3, []float64{0, -2, -4}),
 		},
 		{
-			a:     NewDense(3, 1, []float64{0, 1, 2}).ColView(0),
+			a:     NewDense(3, 1, []float64{0, 1, 2}).ColView(0).(*VecDense),
```

Comment:
ColViewVecDense (and below)

Reply:
Please remove the type assertions and make the `a` field a `Vector`. Then do the assertion at the call site.